packages/cli/src/task-list.ts

```diff
@@ -1,66 +1,94 @@
 import * as pMap from 'p-map';
 
-import { Task, TaskConstructor, TaskResult } from '@checkup/core';
+import { Category, Task, TaskClassification, TaskName, TaskResult } from '@checkup/core';
+
+import PriorityMap from './priority-map';
 
 /**
  * @class TaskList
  *
  * Represents a collection of tasks to run.
  */
 export default class TaskList {
-  private defaultTasks: Task[];
+  private _categories: Map<Category, PriorityMap>;
+
+  get categories() {
+    return this._categories;
+  }
 
-  /**
-   *
-   * @param results {TaskResult[]} the results object that aggregates data together for output.
-   */
   constructor() {
-    this.defaultTasks = [];
+    this._categories = new Map<Category, PriorityMap>([
+      [Category.Core, new PriorityMap()],
+      [Category.Migration, new PriorityMap()],
+    ]);
   }
 
   /**
-   * @method addDefault
+   * @method registerTask
    *
    * Adds a default task to the task list, which is executed as part of checkup.
-   *
-   * @param taskConstructor {TaskConstructor} a constructor representing a Task class
+   * @param taskName {TaskName}
+   * @param task {Task}
    */
-  addTask(taskConstructor: TaskConstructor, cliArguments: any) {
-    this.defaultTasks.push(new taskConstructor(cliArguments));
+  registerTask(taskName: TaskName, task: Task, taskClassification: TaskClassification) {
```

Comment:
I recommend adding a default classification that way implementors can modify if they dont want the default.

```suggestion
registerTask(
  taskName: TaskName,
  task: Task,
  taskClassification: TaskClassification = {
    category: Category.Insights
    priority: Priority.Medium
  }
) {
```
see comment about `Insights`

Reply:
I think you'd _always_ want to define the classification. I would actually say we should enforce this, as it doesn't make sense to have a default, as it essentially has no meaning at that point. Thoughts?

Reply:
I don't think it has no meaning if you don't define it, it just means that you are ok with wherever the framework places your output.

I don't feel very strongly about this, if you are on the firming on the side of no defaults, then we can go that way.

On another note, as I review this I think we are going to need to do a pass over our code once we get closer to launch and throw errors in places where crucial data is not provided, we cannot guarantee our customers are using TS and having better error messages than "x is undefined" will help out a ton.